tools/provision/main.go

```diff
@@ -0,0 +1,264 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"os"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	sdk "github.com/mainflux/mainflux/sdk/go"
+)
+
+type connection struct {
+	ChannelID string
+	ThingID   string
+	ThingKey  string
+	MtlsCert  string
+	MtlsKey   string
+}
+type Connections struct {
+	Connection []connection
+}
+
+const (
+	CRT_LOCATION  = "certs"
+	KEY           = "default"
+	Organization  = "Mainflux"
+	OU            = "mainflux"
+	EA            = "info@mainflux.com"
+	CN            = "localhost"
+	CRT_FILE_NAME = "thing"
+	rsaBits       = 4096
+	daysValid     = 730
+)
+
+func main() {
+	var (
+		host     = flag.String("host", "http://localhost", "Mainflux host address")
+		username = flag.String("username", "", "mainflux user")
+		password = flag.String("password", "", "mainflux user password")
+		num      = flag.Int("num", 10, "number of created channels")
+		ssl      = flag.Bool("ssl", true, "create thing certs")
+		ca       = flag.String("ca", "ca.crt", "CA file for creating things certs")
+		prefix   = flag.String("prefix", "test", "name prefix for created channels and things")
+	)
+
+	flag.Parse()
+
+	msgContentType := string(sdk.CTJSONSenML)
+	sdkConf := sdk.Config{
+		BaseURL:           *host,
+		ReaderURL:         "http://localhost:8905",
+		ReaderPrefix:      "",
+		UsersPrefix:       "",
+		ThingsPrefix:      "",
+		HTTPAdapterPrefix: "http",
+		MsgContentType:    sdk.ContentType(msgContentType),
+		TLSVerification:   false,
+	}
+
+	s := sdk.NewSDK(sdkConf)
+
+	user := sdk.User{
+		Email:    *username,
+		Password: *password,
+	}
+
+	token, err := s.CreateToken(user)
+	if err != nil {
+		log.Fatalf(err.Error())
+		return
+	}
+
+	things := []sdk.Thing{}
+	channels := []sdk.Channel{}
+	connections := Connections{Connection: []connection{}}
+	var caCert *x509.Certificate
+	if *ssl {
+
+		b, err := ioutil.ReadFile(*ca)
+		if err != nil {
+			log.Fatalf("Failed to load CA cert")
+		}
+		block, _ := pem.Decode(b)
+		caCert, err = x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			log.Fatalf("Failed to load CA cert")
+		}
+		log.Printf("Loaded CA cert %s", caCert.Subject.CommonName)
+	}
+
+	for i := 0; i < *num; i++ {
+
+		m, err := createThing(s, fmt.Sprintf("%s-thing-%d", prefix, i), token)
+		if err != nil {
+			log.Println("Failed to create thing")
+			return
+		}
+
+		ch, err := createChannel(s, fmt.Sprintf("%s-channel-%d", prefix, i), token)
+
+		if err := s.ConnectThing(m.ID, ch.ID, token); err != nil {
+			log.Println("Failed to create thing")
+			return
+		}
+
+		channels = append(channels, ch)
+		m, err = s.Thing(m.ID, token)
+		things = append(things, m)
+		cert := ""
+		key := ""
+		if *ssl {
+			// 		thing_cert:
```

Comment:
What are these tabs in the comment? Can you please fix this so that it is nice formatted.